main.ts

```diff
@@ -147,75 +167,88 @@ class SettingsTab extends PluginSettingTab {
+        this.addTemplateToggle(file);
+      }
     }
-    const templaterTemplates = this.getTemplateFiles(coreTemplateFolder, coreTemplateFolderPath).map((e): TemplateFile => {
-      return { path: e, plugin: "templater" };
-    });
+    if (this.plugin.activePlugins.includes('core')) {
+      // core templates plugin
+      containerEl.createEl("h3", {
+        text: "Templates defined by the core Templates plugin",
+      });
+      const coreTemplates = this.getTemplateFiles(this.plugin.coreTemplateFolder, this.plugin.coreTemplateFolderPath).map((e): TemplateFile => {
+        return { path: e, plugin: "core" };
+      });
+      for (const file of coreTemplates) {
+        this.addTemplateToggle(file);
+      }
+    }
+  }
 
-    for (const file of templaterTemplates) {
-      this.addTemplateToggle(file);
+  templateIsEnabled(file: TemplateFile): boolean {
+    switch(file.plugin) {
+      case 'core':
+        return this.plugin.settings.coreFiles.contains(file.path);
+      case 'templater-obsidian':
+        return this.plugin.settings.templaterFiles.contains(file.path);
+      default:
+        console.log(file.path + ' is associated with an unknown plugin');
+        return false;
     }
   }
 
   addTemplateToggle(file: TemplateFile) {
     new Setting(this.containerEl)
       .setName(file.path.replace(".md", ""))
       .addToggle(cb => cb
-        .setValue(file.plugin === "core" ? this.plugin.settings.files.contains(file.path) : this.plugin.settings.templaterFiles.contains(file.path))
+        .setValue(this.templateIsEnabled(file))
         .onChange((value) => this.onToggleChange(value, file)));
   }
   onToggleChange(value: boolean, file: TemplateFile) {
     if (value) {
-      if (file.plugin === "core") {
-        this.plugin.settings.files.push(file.path);
-      } else {
-        this.plugin.settings.templaterFiles.push(file.path);
+      switch(file.plugin) {
+        case 'core':
+          this.plugin.settings.coreFiles.push(file.path);
+          break;
+        case 'templater-obsidian':
+          this.plugin.settings.templaterFiles.push(file.path);
+          break;
+        default:
+          console.log(file.path + ' is associated with an unknown plugin');
+          return;
       }
       this.plugin.pushCommand(file);
     } else {
-      if (file.plugin === "core") {
-        this.plugin.settings.files.remove(file.path);
-      } else {
-        this.plugin.settings.templaterFiles.remove(file.path);
+      switch(file.plugin) {
+        case 'core':
+          this.plugin.settings.coreFiles.remove(file.path);
+          break;
+        case 'templater-obsidian':
+          this.plugin.settings.templaterFiles.remove(file.path);
+          break;
+        default:
+          console.log(file.path + ' is associated with an unknown plugin');
+          return;
       }
       (this.plugin.app as any).commands.removeCommand(`${this.plugin.manifest.id}:${file.plugin}:${file.path}`);

```

Comment:
Because of the backwards compatibility, the id of the core template command does not contain the plugin name. So unloading a core template does not work.